modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -165,14 +164,20 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
         line(s"""val name: String = "$originalName""""),
         line(s"""val version: String = "$version""""),
         newline,
-        block(
-          s"def endpoint[I, E, O, SI, SO](op : $opTraitName[I, E, O, SI, SO]) = op match"
-        ) {
-          ops.map {
-            case op if op.input != Type.unit =>
-              s"case ${op.name}(input) => (input, ${op.name})"
-            case op =>
-              s"case ${op.name}() => ((), ${op.name})"
+        if (ops.isEmpty) {
+          line(
+            s"""def endpoint[I, E, O, SI, SO](op : $opTraitName[I, E, O, SI, SO]) = sys.error("impossible")"""
```

Comment:
Because the sealed trait is unpopulated, it is effectively a phantom type, therefore it is literally impossible under normal circumstances (unless runtime reflection is involved, or black magic) for this method to be called.